### ISSUE TITLE
copy with snapshots from 2.x to 3.x fails: proto: bad wiretype for field main.MigrationHeader.Criu: got wiretype 2, want 0

### ISSUE THREAD

**Issue by fwaggle** [lxc/lxd#4780]
lxclab4 is Ubuntu Xenial with the shipped LXD package installed (2.0, but 2.2 fails as well). lxclab3 is Ubuntu Bionic with LXD 3.2 built from git (commit 140c4a6f373aacc97f0ba00bf31316894fa32989), but it fails on LXD3.2 from snap store as well. Both using ZFS for storage (full `lxc info` below):

```
$ lxc info | grep version
api_version: "1.0"
  driver_version: 2.0.8
  kernel_version: 4.4.0-130-generic
  server_version: 2.0.11
  storage_version: "5"
$ lxc info lxclab3: | grep version
api_version: "1.0"
  driver_version: 3.0.1
  kernel_version: 4.15.0-23-generic
  server_version: "3.2"
  storage_version: 0.7.5-1ubuntu15
```

On LXD 2, I create an alpine container and can copy it to LXD3.2 without issues:

```
$ lxc init images:alpine/3.7
Creating the container
Container name is: equal-aardvark
$ lxc list | grep equal-aardvark
| equal-aardvark | STOPPED |      |      | PERSISTENT | 0         |
$ lxc copy equal-aardvark lxclab3:
$ lxc delete lxclab3:equal-aardvark
$
```

However if a snapshot exists, it hangs without activity/failure:

```
$ lxc snapshot equal-aardvark snap1
$ lxc list | grep equal-aardvark
| equal-aardvark | STOPPED |      |      | PERSISTENT | 1         |
$ lxc copy equal-aardvark lxclab3:
[hang]
```

I have full copies of `lxc monitor` attached below, but this seems to be the issue (the monitor attached to lxd2.x, the one on 3.x reports nothing super interesting):

```
metadata:
  context: {}
  level: dbug
  message: 'Failure for websocket operation: cd5b8482-b79c-4b7b-a712-c588fc606687:
    proto: bad wiretype for field main.MigrationHeader.Criu: got wiretype 2, want
    0'
timestamp: 2018-07-13T08:28:55.895758535+10:00
type: logging


metadata:
  class: websocket
  created_at: 2018-07-13T08:28:55.046809597+10:00
  err: 'proto: bad wiretype for field main.MigrationHeader.Criu: got wiretype 2, want
    0'
  id: cd5b8482-b79c-4b7b-a712-c588fc606687
  may_cancel: false
  metadata:
    control: 17dcdca999c1accc353271507f1a82d5f22b68d67d432a0cc7eee12c2af87dba
    fs: b28815a0f608fa7ca69c2641d51fe850879475c57fc01477ef1e713c61037f4e
  resources:
    containers:
    - /1.0/containers/equal-aardvark
  status: Failure
  status_code: 400
  updated_at: 2018-07-13T08:28:55.046809597+10:00
timestamp: 2018-07-13T08:28:55.895788232+10:00
type: operation
```

If you terminate the `lxc` client, you can copy/move other containers around still. If you copy the snapshot to a container on the 2.x LXD, you can copy the resulting container to 3.x without issue.

# lxc info (2.x)

```
config:
  core.https_address: '[::]:8443'
  core.trust_password: true
  storage.zfs_pool_name: lxd
api_extensions:
- id_map
- id_map_base
- resource_limits
api_status: stable
api_version: "1.0"
auth: trusted
auth_methods: []
public: false
environment:
  addresses:
  - 10.0.4.4:8443
  - 10.17.33.1:8443
  architectures:
  - x86_64
  - i686
  certificate: |
    -----BEGIN CERTIFICATE-----
    MIIFQzCCAyugAwIBAgIQSY2STwUa3UzACvb9Wed9LDANBgkqhkiG9w0BAQsFADA1
    MRwwGgYDVQQKExNsaW51eGNvbnRhaW5lcnMub3JnMRUwEwYDVQQDDAxyb290QGx4
    Y2xhYjQwHhcNMTgwNzExMjIyMjU4WhcNMjgwNzA4MjIyMjU4WjA1MRwwGgYDVQQK
    ExNsaW51eGNvbnRhaW5lcnMub3JnMRUwEwYDVQQDDAxyb290QGx4Y2xhYjQwggIi
    MA0GCSqGSIb3DQEBAQUAA4ICDwAwggIKAoICAQDmadI4DU4hJIcS2IMl8rCyeZOd
    jVmQ9YGH97xwx98eCKHwPvdKfTWTbTeBjcpACE7iggJOZA/sjHjL8E/NxA61aKKq
    TMzmoCxFNvaL57y6t3hpYpEoNBchcoKexUyplOJPvA4SX4OC7lb+exkOme1hLJKd
    pL08ric7n5sisUwQDDpcGQvZG1qUDt31F581yhaj5iiFrok8xw91nbZqPqBszF5r
    AjXLN+qcQvAUPDKuJvGZH92tGioUdXXbhDYS04/63OyL/k9NXMPHgkxH4qg9TD2g
    junu9FmwHEwJbSYHbHQr7+WEm/G5HJ5pXEoaamwJA1ooxKczk7Kb0HW1IUKwN4NM
    9YT1L9m5jVlW+Zj2In2C2XaIcVMQX+i06G9xhCDKY67BZQZzpvupSBLgIJzR6Gva
    jUvL6fw6bIIeWe+chzG1x6RYUPXTipTmJmLEXc9frguzAs8xAHxLV4F6tw+gIkBM
    EMFcEUPQicjQaa8sTr+4e1XlbNz0JqtbQykhLQ6/aFusTuZJ4jEU8MySSnKvFu2Q
    FIKbG+qHfuKSMk07k100JsaRgYpfaFLHIJyEv9RkzJEXesu0fSgbrRhi23ZqAolP
    JfHsp9kq+obUJ06ZP7tQGxLw+QPpoQq9cjlpmyrHRev2a1l9fvu5fBWFTnbNz5n+
    zRPkIdKtz24qZCPUFwIDAQABo08wTTAOBgNVHQ8BAf8EBAMCBaAwEwYDVR0lBAww
    CgYIKwYBBQUHAwEwDAYDVR0TAQH/BAIwADAYBgNVHREEETAPggdseGNsYWI0hwQK
    AAQEMA0GCSqGSIb3DQEBCwUAA4ICAQA+U6DGhyT1T3Nkk2kwtM+XjYHMMbIlcjie
    F9656lVE4Xju9moNz8teyGkl6egAo2coqU6ySZBtgBhp7B0yvZG3DMmUFvbCDzlq
    ObCPI5cBjBfo1yZGms50/oWjj7f/Pu1doB+XQnT+YNxw5Kdv/dQkyx5zKeSTmOk/
    We56GzSdlAs+JeCne+RGuNIdCKh9c1OZBUjyITyZFb3ltoS1cVtsUW5wNiWGLLew
    Mu5uCpe+9X0/KIm0oANysRoP75aLku7SZpw0h7hKOx9b0YDqPBPWU3qlsH+SJft0
    UZixobSOLF9iljEJNse11yj6oGAA1lk/eryff1F+b15Hx57hxSOlEctJx3PXKUN4
    tJxjeMPXifEV5/ob5vIozVpUEQJ8jFBkYW3rOTBEIOBzFDRvsLfRrTl0LOzyxE1z
    yUWKFKXWOO/gV9Vu87DFHFa7ad2RtTnAdOkM0L+crHS0M5MCYBfOO4MOhaU6+ziB
    LPBlc2HkxkOlYXITNUs5E69KjdzW/4YPqSqsoR0f1q7TM8oaUBmRLXQhkLQv3Tvp
    wk02D4RGMq/U5pIA86vRa9cq+HfQwwT9fO2/irmPoXOmwp2BnEr7KNZK9dw33xzs
    Qa9vRCQBtHT/sdsrfIucuOErUzJYl6YOODScgXamD4DrNSdPCcNwWmRn2F8rHrLc
    JGXrynlReg==
    -----END CERTIFICATE-----
  certificate_fingerprint: 2ea7ff3d1a7bbc854a6ab67817387489e5d33e1e06e4626539958ad10bed14df
  driver: lxc
  driver_version: 2.0.8
  kernel: Linux
  kernel_architecture: x86_64
  kernel_version: 4.4.0-130-generic
  server: lxd
  server_pid: 2470
  server_version: 2.0.11
  storage: zfs
  storage_version: "5"
```

# lxc info (3.2)

```
config:
  core.debug_address: 127.0.0.1:8080
  core.https_address: '[::]:8443'
  core.trust_password: true
api_extensions:
- storage_zfs_remove_snapshots
- container_host_shutdown_timeout
- container_stop_priority
- container_syscall_filtering
- auth_pki
- container_last_used_at
- etag
- patch
- usb_devices
- https_allowed_credentials
- image_compression_algorithm
- directory_manipulation
- container_cpu_time
- storage_zfs_use_refquota
- storage_lvm_mount_options
- network
- profile_usedby
- container_push
- container_exec_recording
- certificate_update
- container_exec_signal_handling
- gpu_devices
- container_image_properties
- migration_progress
- id_map
- network_firewall_filtering
- network_routes
- storage
- file_delete
- file_append
- network_dhcp_expiry
- storage_lvm_vg_rename
- storage_lvm_thinpool_rename
- network_vlan
- image_create_aliases
- container_stateless_copy
- container_only_migration
- storage_zfs_clone_copy
- unix_device_rename
- storage_lvm_use_thinpool
- storage_rsync_bwlimit
- network_vxlan_interface
- storage_btrfs_mount_options
- entity_description
- image_force_refresh
- storage_lvm_lv_resizing
- id_map_base
- file_symlinks
- container_push_target
- network_vlan_physical
- storage_images_delete
- container_edit_metadata
- container_snapshot_stateful_migration
- storage_driver_ceph
- storage_ceph_user_name
- resource_limits
- storage_volatile_initial_source
- storage_ceph_force_osd_reuse
- storage_block_filesystem_btrfs
- resources
- kernel_limits
- storage_api_volume_rename
- macaroon_authentication
- network_sriov
- console
- restrict_devlxd
- migration_pre_copy
- infiniband
- maas_network
- devlxd_events
- proxy
- network_dhcp_gateway
- file_get_symlink
- network_leases
- unix_device_hotplug
- storage_api_local_volume_handling
- operation_description
- clustering
- event_lifecycle
- storage_api_remote_volume_handling
- nvidia_runtime
- container_mount_propagation
- container_backup
- devlxd_images
- container_local_cross_pool_handling
- proxy_unix
- proxy_udp
- clustering_join
- proxy_tcp_udp_multi_port_handling
- network_state
- proxy_unix_dac_properties
- container_protection_delete
- unix_priv_drop
- pprof_http
api_status: stable
api_version: "1.0"
auth: trusted
auth_methods:
- tls
public: false
environment:
  addresses:
  - 10.0.4.3:8443
  - 10.220.96.1:8443
  architectures:
  - x86_64
  - i686
  certificate: |
    -----BEGIN CERTIFICATE-----
    MIIFRDCCAyygAwIBAgIRAIfGQ7WTX4LsllOytxLpS7swDQYJKoZIhvcNAQELBQAw
    NTEcMBoGA1UEChMTbGludXhjb250YWluZXJzLm9yZzEVMBMGA1UEAwwMcm9vdEBs
    eGNsYWIzMB4XDTE4MDcxMjAwNDcwMloXDTI4MDcwOTAwNDcwMlowNTEcMBoGA1UE
    ChMTbGludXhjb250YWluZXJzLm9yZzEVMBMGA1UEAwwMcm9vdEBseGNsYWIzMIIC
    IjANBgkqhkiG9w0BAQEFAAOCAg8AMIICCgKCAgEAyw9/reU8Pz98rKvtHsi1yA9N
    p8Dcime/o9n0z6YYOEPurwBRN6P/h8xnSQHdXXZMQqfvSr35ffbx/KKDToYV014s
    eP0i+a1QItPul/DtWi0OpoaEfZjXuthtP+THIN5DJyn7FJynPO6iD3TOgwaAcgTx
    bG/cvNGGSfk2asP9Vv3u/Wu4IBTo1yy6nOYOtgxzwISYSuWKJpz9jOPx1wMWCED7
    rvGpXn/sce3tNu7nR9XTyafaxJNP1hyMrKnh5YxcNM4UmaijP9g4wUzEDyecFs70
    QJsMRoKbaqTJtLoCD6h09mNoEMXB73mzliJGEqzmtrHpPk6VVplAn3+C9hb07V7b
    OLBG84hKx0fEWCCte2PQWtnxumXLn62ohomXTIxiZLpLMQXKrJxT9iiG/HM6s6v3
    zTyFkzMNV4fPOGMloCSL3ufP9T8iwIj2X6+86AbPUGI0/mfH4KPCNNqza490Nv9l
    b6pSo5HD5NMaqmBvVM8VEzmz33MSPWLWklfgnJDLNBjveGmR+UmHqwl7ESg69e6n
    Ok2va7aq1MDGug0uUMXSRaizHiV3sF5smajYxeirZ8RC8GMaayJLD25CWbbyRH+d
    WSTrbJcZOVrG8nUGyCxocYKMKeQRMSuR9a5w0ndRpoNaed4thiwwYQkG7hX115je
    1J29+u1+EENVVcS7wxkCAwEAAaNPME0wDgYDVR0PAQH/BAQDAgWgMBMGA1UdJQQM
    MAoGCCsGAQUFBwMBMAwGA1UdEwEB/wQCMAAwGAYDVR0RBBEwD4IHbHhjbGFiM4cE
    CgAEAzANBgkqhkiG9w0BAQsFAAOCAgEAjPVWLYxNYedPtdaQkXDs6FWcpSCFYB8L
    kn6krwKwbqKpgS3jhz4FZD6Mtq+IpWgWYIew8NV+NTrjyT44AcF620SWa5zy7XKS
    7k2ou3+zH0VfGTvKmNnFMmhFlxUfh0LrR312vsCIErb35dDrvMplZVZBIjAbkQ83
    lcCAqJfMryGhyMNhGr1P5JHA26KN8IYznMH6e5niXL7TLXk7r3cUw5MSyGC3IIUY
    +2B1Uw/ZErPMYmk6SRTotNraKcrqGvfSAE7xR/E4ENDwhp+5rQrPoe+id/JkpGVr
    d+IRW8t+RsgYtfhR8Dva5kDDu0vJseuZGf6x62V5QDGeUBnaSZe7ZQ6X5VzP8fx7
    cNjQzzvVoakVTjtna89aXMvun+9RkKP5aUD6jcpobBGDWao0+EGxnkc9H8VIZh2Q
    UF2d0t3Ee/ivzfumoucRn83Ds+NJ1yB7vuYKJc4W/wpw1I7LTCpOSw4HpUrvZ7ES
    12Agn2FLjoM2zk1qtQsv157qtpDHcfHw9mVpJaGuUsWJs7cc0c5CSC0dcGpyb7UB
    HAit2TAHr+wuYxix0cRN1TBDO7r5SYPfrzyGMFMEJpRT7K/oFB+Q1URjG+0WCk04
    gZVFJJQ6l5aVky68cqKvIev6sECSi299kYGOzfkUw8jz7kbrJf7SwuIXQomwdGSr
    onTIhyfyxnU=
    -----END CERTIFICATE-----
  certificate_fingerprint: 7d8403fc89a0ed261a1190feddbb02f8e9b50785e5b21b3cc40233ba4ebe786c
  driver: lxc
  driver_version: 3.0.1
  kernel: Linux
  kernel_architecture: x86_64
  kernel_version: 4.15.0-23-generic
  server: lxd
  server_pid: 3689
  server_version: "3.2"
  storage: zfs
  storage_version: 0.7.5-1ubuntu15
```

# lxc monitor (2.x)

```
metadata:
  context: {}
  level: dbug
  message: 'New events listener: 7ce2d2e7-3bce-479a-a6b6-bc95dadc9043'
timestamp: 2018-07-13T08:28:52.98966989+10:00
type: logging


metadata:
  context:
    ip: '@'
    method: GET
    url: /1.0
  level: dbug
  message: handling
timestamp: 2018-07-13T08:28:54.963484263+10:00
type: logging


metadata:
  context:
    ip: '@'
    method: GET
    url: /1.0/containers/equal-aardvark
  level: dbug
  message: handling
timestamp: 2018-07-13T08:28:55.042562574+10:00
type: logging


metadata:
  context:
    ip: '@'
    method: GET
    url: /1.0/events
  level: dbug
  message: handling
timestamp: 2018-07-13T08:28:55.045016712+10:00
type: logging


metadata:
  context: {}
  level: dbug
  message: 'New events listener: 8be3c0b4-6063-44dc-8e6b-add4eb32f576'
timestamp: 2018-07-13T08:28:55.045051073+10:00
type: logging


metadata:
  context:
    ip: '@'
    method: POST
    url: /1.0/containers/equal-aardvark
  level: dbug
  message: handling
timestamp: 2018-07-13T08:28:55.045691618+10:00
type: logging


metadata:
  class: websocket
  created_at: 2018-07-13T08:28:55.046809597+10:00
  err: ""
  id: cd5b8482-b79c-4b7b-a712-c588fc606687
  may_cancel: false
  metadata:
    control: 17dcdca999c1accc353271507f1a82d5f22b68d67d432a0cc7eee12c2af87dba
    fs: b28815a0f608fa7ca69c2641d51fe850879475c57fc01477ef1e713c61037f4e
  resources:
    containers:
    - /1.0/containers/equal-aardvark
  status: Running
  status_code: 103
  updated_at: 2018-07-13T08:28:55.046809597+10:00
timestamp: 2018-07-13T08:28:55.046860804+10:00
type: operation


metadata:
  class: websocket
  created_at: 2018-07-13T08:28:55.046809597+10:00
  err: ""
  id: cd5b8482-b79c-4b7b-a712-c588fc606687
  may_cancel: false
  metadata:
    control: 17dcdca999c1accc353271507f1a82d5f22b68d67d432a0cc7eee12c2af87dba
    fs: b28815a0f608fa7ca69c2641d51fe850879475c57fc01477ef1e713c61037f4e
  resources:
    containers:
    - /1.0/containers/equal-aardvark
  status: Pending
  status_code: 105
  updated_at: 2018-07-13T08:28:55.046809597+10:00
timestamp: 2018-07-13T08:28:55.046830301+10:00
type: operation


metadata:
  context: {}
  level: dbug
  message: 'Started websocket operation: cd5b8482-b79c-4b7b-a712-c588fc606687'
timestamp: 2018-07-13T08:28:55.046846142+10:00
type: logging


metadata:
  context: {}
  level: dbug
  message: 'New websocket operation: cd5b8482-b79c-4b7b-a712-c588fc606687'
timestamp: 2018-07-13T08:28:55.046816087+10:00
type: logging


metadata:
  context:
    ip: 10.0.4.3:57742
    url: /1.0/operations/cd5b8482-b79c-4b7b-a712-c588fc606687/websocket?secret=17dcdca999c1accc353271507f1a82d5f22b68d67d432a0cc7eee12c2af87dba
  level: dbug
  message: allowing untrusted GET
timestamp: 2018-07-13T08:28:55.875314579+10:00
type: logging


metadata:
  context: {}
  level: dbug
  message: 'Connected websocket operation: cd5b8482-b79c-4b7b-a712-c588fc606687'
timestamp: 2018-07-13T08:28:55.875343758+10:00
type: logging


metadata:
  context: {}
  level: dbug
  message: 'Handled websocket operation: cd5b8482-b79c-4b7b-a712-c588fc606687'
timestamp: 2018-07-13T08:28:55.875459302+10:00
type: logging


metadata:
  context: {}
  level: dbug
  message: 'Connected websocket operation: cd5b8482-b79c-4b7b-a712-c588fc606687'
timestamp: 2018-07-13T08:28:55.88959426+10:00
type: logging


metadata:
  context:
    ip: 10.0.4.3:57744
    url: /1.0/operations/cd5b8482-b79c-4b7b-a712-c588fc606687/websocket?secret=b28815a0f608fa7ca69c2641d51fe850879475c57fc01477ef1e713c61037f4e
  level: dbug
  message: allowing untrusted GET
timestamp: 2018-07-13T08:28:55.889563889+10:00
type: logging


metadata:
  context:
    container: equal-aardvark
    driver: storage/zfs
  level: dbug
  message: ContainerStart
timestamp: 2018-07-13T08:28:55.890850158+10:00
type: logging


metadata:
  context: {}
  level: dbug
  message: 'Handled websocket operation: cd5b8482-b79c-4b7b-a712-c588fc606687'
timestamp: 2018-07-13T08:28:55.889694905+10:00
type: logging


metadata:
  context:
    container: equal-aardvark
    driver: storage/zfs
  level: dbug
  message: MigrationSource
timestamp: 2018-07-13T08:28:55.892747917+10:00
type: logging


metadata:
  context:
    container: equal-aardvark
    driver: storage/zfs
  level: dbug
  message: ContainerStop
timestamp: 2018-07-13T08:28:55.895711411+10:00
type: logging


metadata:
  context: {}
  level: dbug
  message: 'Failure for websocket operation: cd5b8482-b79c-4b7b-a712-c588fc606687:
    proto: bad wiretype for field main.MigrationHeader.Criu: got wiretype 2, want
    0'
timestamp: 2018-07-13T08:28:55.895758535+10:00
type: logging


metadata:
  class: websocket
  created_at: 2018-07-13T08:28:55.046809597+10:00
  err: 'proto: bad wiretype for field main.MigrationHeader.Criu: got wiretype 2, want
    0'
  id: cd5b8482-b79c-4b7b-a712-c588fc606687
  may_cancel: false
  metadata:
    control: 17dcdca999c1accc353271507f1a82d5f22b68d67d432a0cc7eee12c2af87dba
    fs: b28815a0f608fa7ca69c2641d51fe850879475c57fc01477ef1e713c61037f4e
  resources:
    containers:
    - /1.0/containers/equal-aardvark
  status: Failure
  status_code: 400
  updated_at: 2018-07-13T08:28:55.046809597+10:00
timestamp: 2018-07-13T08:28:55.895788232+10:00
type: operation
```

# lxc monitor (3.2)
```
metadata:
  context: {}
  level: dbug
  message: 'Disconnected event listener: 80dccccd-6ef8-4a85-b008-6ab2cfa24a2a'
timestamp: "2018-07-12T22:28:51.6785021Z"
type: logging


metadata:
  context: {}
  level: dbug
  message: 'New event listener: e57dc385-ba06-4e07-9906-040f642c6d79'
timestamp: "2018-07-12T22:28:51.678441256Z"
type: logging


metadata:
  context:
    ip: 10.0.4.4:36470
    method: GET
    url: /1.0
  level: dbug
  message: handling
timestamp: "2018-07-12T22:28:54.994149027Z"
type: logging


metadata:
  context:
    k: "0"
  level: dbug
  message: Found cert
timestamp: "2018-07-12T22:28:54.994098245Z"
type: logging


metadata:
  context:
    k: "0"
  level: dbug
  message: Found cert
timestamp: "2018-07-12T22:28:54.995138539Z"
type: logging


metadata:
  context:
    k: "0"
  level: dbug
  message: Found cert
timestamp: "2018-07-12T22:28:55.063012395Z"
type: logging


metadata:
  context:
    ip: 10.0.4.4:36472
    method: GET
    url: /1.0/events
  level: dbug
  message: handling
timestamp: "2018-07-12T22:28:55.063134353Z"
type: logging


metadata:
  context: {}
  level: dbug
  message: 'New event listener: d8d7ae37-4f3f-4772-b77d-13e9cb646ba3'
timestamp: "2018-07-12T22:28:55.063390108Z"
type: logging


metadata:
  context:
    k: "0"
  level: dbug
  message: Found cert
timestamp: "2018-07-12T22:28:55.088258661Z"
type: logging


metadata:
  context: {}
  level: dbug
  message: Responding to container create
timestamp: "2018-07-12T22:28:55.088461319Z"
type: logging


metadata:
  context:
    ip: 10.0.4.4:36474
    method: POST
    url: /1.0/containers
  level: dbug
  message: handling
timestamp: "2018-07-12T22:28:55.088357662Z"
type: logging


metadata:
  context: {}
  level: dbug
  message: No valid storage pool in the container's local root disk device and profiles
    found
timestamp: "2018-07-12T22:28:55.090358331Z"
type: logging


metadata:
  context: {}
  level: dbug
  message: Initializing a ZFS driver
timestamp: "2018-07-12T22:28:55.092385118Z"
type: logging


metadata:
  context: {}
  level: dbug
  message: 'Database error: &errors.errorString{s:"sql: no rows in result set"}'
timestamp: "2018-07-12T22:28:55.092895066Z"
type: logging


metadata:
  context:
    ephemeral: "false"
    name: equal-aardvark
  level: info
  message: Creating container
timestamp: "2018-07-12T22:28:55.149624121Z"
type: logging


metadata:
  context: {}
  level: dbug
  message: Initializing a ZFS driver
timestamp: "2018-07-12T22:28:55.157285079Z"
type: logging


metadata:
  action: container-updated
  source: /1.0/containers/equal-aardvark
timestamp: "2018-07-12T22:28:55.171113756Z"
type: lifecycle


metadata:
  action: container-updated
  source: /1.0/containers/equal-aardvark
timestamp: "2018-07-12T22:28:55.180665814Z"
type: lifecycle


metadata:
  action: container-updated
  source: /1.0/containers/equal-aardvark
timestamp: "2018-07-12T22:28:55.193293539Z"
type: lifecycle


metadata:
  context: {}
  level: dbug
  message: dbQueryScan() took 2 ms
timestamp: "2018-07-12T22:28:55.197752621Z"
type: logging


metadata:
  context:
    ephemeral: "false"
    name: equal-aardvark
  level: info
  message: Created container
timestamp: "2018-07-12T22:28:55.489132479Z"
type: logging


metadata:
  action: container-created
  source: /1.0/containers/equal-aardvark
timestamp: "2018-07-12T22:28:55.489153281Z"
type: lifecycle


metadata:
  context: {}
  level: dbug
  message: Creating empty ZFS storage volume for container "equal-aardvark" on storage
    pool "default"
timestamp: "2018-07-12T22:28:55.489168897Z"
type: logging


metadata:
  context: {}
  level: dbug
  message: Mounting ZFS storage volume for container "equal-aardvark" on storage pool
    "default"
timestamp: "2018-07-12T22:28:55.58466843Z"
type: logging


metadata:
  context: {}
  level: dbug
  message: Created empty ZFS storage volume for container "equal-aardvark" on storage
    pool "default"
timestamp: "2018-07-12T22:28:55.584624886Z"
type: logging


metadata:
  context: {}
  level: dbug
  message: Mounted ZFS storage volume for container "equal-aardvark" on storage pool
    "default"
timestamp: "2018-07-12T22:28:55.586657235Z"
type: logging


metadata:
  context: {}
  level: dbug
  message: Unmounting ZFS storage volume for container "equal-aardvark" on storage
    pool "default"
timestamp: "2018-07-12T22:28:55.72507021Z"
type: logging


metadata:
  context:
    name: equal-aardvark
  level: warn
  message: Unable to update backup.yaml at this time
timestamp: "2018-07-12T22:28:55.724931497Z"
type: logging


metadata:
  action: container-updated
  source: /1.0/containers/equal-aardvark
timestamp: "2018-07-12T22:28:55.724992012Z"
type: lifecycle


metadata:
  context: {}
  level: dbug
  message: Mounting ZFS storage volume for container "equal-aardvark" on storage pool
    "default"
timestamp: "2018-07-12T22:28:55.759080984Z"
type: logging


metadata:
  context: {}
  level: dbug
  message: Unmounted ZFS storage volume for container "equal-aardvark" on storage
    pool "default"
timestamp: "2018-07-12T22:28:55.759046606Z"
type: logging


metadata:
  context: {}
  level: dbug
  message: Unmounting ZFS storage volume for container "equal-aardvark" on storage
    pool "default"
timestamp: "2018-07-12T22:28:55.761226414Z"
type: logging


metadata:
  context:
    name: equal-aardvark
  level: warn
  message: Unable to update backup.yaml at this time
timestamp: "2018-07-12T22:28:55.761205574Z"
type: logging


metadata:
  context: {}
  level: dbug
  message: Mounted ZFS storage volume for container "equal-aardvark" on storage pool
    "default"
timestamp: "2018-07-12T22:28:55.761140527Z"
type: logging


metadata:
  context: {}
  level: dbug
  message: Unmounted ZFS storage volume for container "equal-aardvark" on storage
    pool "default"
timestamp: "2018-07-12T22:28:55.787280019Z"
type: logging


metadata:
  context: {}
  level: dbug
  message: 'New task operation: bc243a2e-f137-4e37-9176-10019908088e'
timestamp: "2018-07-12T22:28:55.83982174Z"
type: logging


metadata:
  context: {}
  level: dbug
  message: 'Started task operation: bc243a2e-f137-4e37-9176-10019908088e'
timestamp: "2018-07-12T22:28:55.839964234Z"
type: logging


metadata:
  class: task
  created_at: "2018-07-12T22:28:55.787384571Z"
  description: Creating container
  err: ""
  id: bc243a2e-f137-4e37-9176-10019908088e
  may_cancel: false
  metadata: null
  resources:
    containers:
    - /1.0/containers/equal-aardvark
  status: Running
  status_code: 103
  updated_at: "2018-07-12T22:28:55.787384571Z"
timestamp: "2018-07-12T22:28:55.839982564Z"
type: operation


metadata:
  class: task
  created_at: "2018-07-12T22:28:55.787384571Z"
  description: Creating container
  err: ""
  id: bc243a2e-f137-4e37-9176-10019908088e
  may_cancel: false
  metadata: null
  resources:
    containers:
    - /1.0/containers/equal-aardvark
  status: Pending
  status_code: 105
  updated_at: "2018-07-12T22:28:55.787384571Z"
timestamp: "2018-07-12T22:28:55.839904667Z"
type: operation


metadata:
  context:
    ip: 10.0.4.4:36476
    method: GET
    url: /1.0/operations/bc243a2e-f137-4e37-9176-10019908088e
  level: dbug
  message: handling
timestamp: "2018-07-12T22:28:55.874084564Z"
type: logging


metadata:
  context:
    k: "0"
  level: dbug
  message: Found cert
timestamp: "2018-07-12T22:28:55.874032407Z"
type: logging
```

Hope that's useful!

**Comment by stgraber** [comment]
Hmm, so that's annoying. Basically the problem is that the protobuf wireformat in LXD 2.0.x is broken, some fields have conflicting indices. Newer releases have this bug fixed, but that means that you can't migrate some attributes from LXD 2.0.x to LXD 2.21 or higher (that's my guess as to when the protocol got fixed).

The good news is that the fix is in the LXD 2.0.x branch so LXD 2.0.12 will have this fixed.
The problem though is that then the same will happen if you try to migrate containers from 2.0.11 to 2.0.12, though since it'll be a simple bugfix release, upgrading the source should be less of a problem.

**Comment by stgraber** [comment]
I'm going to test now using what's currently in stable-2.0, if that works fine, I'll close the issue as we technically already have a fix in there. Let me know if a manually built binary would be useful for you and I'll send one to you.

**Comment by stgraber** [comment]
Hmm, so there was that issue and then another one which I've now fixed, but migration still hangs.
I unfortunately won't have much time to look into this more next week, so I'll hand it over to @brauner 

Christian, the initial issue was conflicting field numbers in the protobuf which got fixed in stable-2.0 already. I've then also had to add support for NONE in CriuType to make the parser in 2.0.x understand the header sent by newer releases, but even with that in place, I can't seem to copy snapshots between machines... The copy hangs without any obviously visible error as if the rsync stage never triggers.

My test setup is two 16.04 systems (edfu and nuturo), one running the stock LXD 2.0.11 (dir backend) and the other running the snap (LXD 3.2). I've done a custom build of current stable-2.0 that I put in place on edfu as well as upgraded to current liblxc so that it's easier to test new builds of lxd.

Feel free to just reuse the two machines, if you don't use them, please release them.
